lib/zold/type.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2018 Yegor Bugayenko
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the 'Software'), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# @todo Right now only Score and Http classes have been refactored for using
+# dry-types, even tough the issue has been boosted refactoring the whole project
+# is very cumbersome. Please refer to Score and Http class on how to perform all
+# the changes for the project to adopt dry-types
+require 'dry-types'
+require 'dry-struct'
+
+# HTTP page.
+# Author:: Yegor Bugayenko (yegor256@gmail.com)
+# Copyright:: Copyright (c) 2018 Yegor Bugayenko
+# License:: MIT
+module Types
```

Comment:
I know it's a very technical class, but the usage example of Dry Types suggest using it:

http://dry-rb.org/gems/dry-types/

I am open to other solutions though

Reply:
@yegor256 ping

Reply:
@emilianodellacasa @oltdaniel I also think that we should just use `Dry::Types` and that's it. No need to create our custom module.

Reply:
@yegor256 @oltdaniel I tried removing that module on a single class, and this is the error that is giving to me:

/home/ufi/.rvm/gems/ruby-2.3.3/gems/dry-types-0.13.2/lib/dry/types.rb:141:in `const_missing': dry-types does not define constants for default types. You can access the predefined types with [], e.g. Dry::Types["strict.integer"] or generate a module with types using Dry::Types.module (NameError)

I could of course use the first solution they give, but it will make the code less readable IMHO. Creating the module is just a convenient way to use dry-types.

What do you think?

Reply:
@emilianodellacasa I am currently on vacation. Will be available for testing stuff in 2 weeks.

Reply:
@oltdaniel @yegor256 There is nothing to be tested here: I have just shown to you that the module is required by Dry Types to improve code readability. 

Yegor, if you still want me to get rid of the Type module just take into account what it will mean for readability. Otherwise, just merge the code as it is and let's close this PR.

Reply:
@yegor256 Ping